README.md
# PFUSDK V0.3.0

## 红包功能
[点击这里查看红包功能接入文档](https://github.com/LostChrs/PFUSDK/blob/master/Redpacket.md)

## 更新日志
[点击这里查看版本更新日志](https://github.com/LostChrs/PFUSDK/blob/master/ChangeLog.md)
# 快速接入
将pfusdk文件夹整个拖入cocoscreator工程中，`PfuSdk.js`中包含所有Sdk对外提供的接口，`PfuConfig.js`用来配置Sdk所需参数。将`pfusdk/prefabs/PfuSdk.prefab`直接拖入游戏根节点。注意不要挂载到Canvas下，应该挂载到根节点。

>所有项目均会自动生成常用在线参数，如pfuSdkTestMode(审核模式),pfuSdkShowOpenAds(显示开屏广告)等。审核模式开启时将自动隐藏更多游戏列表，更多游戏按钮等功能。

>注意：配置好PfuConfig参数后，请备份好此配置，防止更新SDK时覆盖了配置文件。
### Config.js中参数配置 (*必接)

| 参数 | 是否必填 | 含义 | 示例/默认值 |
| :------| :------ | :------ | :------|
|version|是|游戏参数版本号，用来控制运营后台参数|1.0.0|
|appId|是| 嘉丰永道appId|1306246|
|wxId|是|微信小游戏Id|wxc3ee0b459142bb1b|
|pfuPrivateKey|是|嘉丰永道分配给合作方的私钥|bp11lflsecbme9itbbf|
|wxVideoId|否|视频广告Id|adunit-c758250e9f75430c|
|wxBannerId|否|banner广告Id|adunit-8c19e95d3c8e9a7a|
|bannerHeight|否|banner高度，范围170~230|170|
|payAppId|否|虚拟支付Id，当接入虚拟支付的时候需要填写|1450018202|
|openInviteListListner|否|开启邀请用户的实时监听|false|
|inviteListUpdateTime|否|更新一次邀请列表频率（秒）|180|

>如果要使用更多游戏列表或更多游戏按钮，需要在发布的微信工程的game.json中添加如下配置

```
"navigateToMiniProgramAppIdList":[
    "wx3e33fef689f472b1",
    "wx2d47467291703ec7",
    "wx716b36314be3fe89",
    "wx8b25b991dcc6edf6",
    "wx99e08aff982f8dde",
    "wx3b33f72ee2ec7bc1",
    "wx989199f7dc0e3a50",
    "wxbbbef0244fa4d4e4",
    "wx87fe1890ea1384a5",
    "wxa0eee1fe564aa730"
  ]
```
### 获取在线参数(*必接)
```
 PfuSdk.Instance.setOnlineParamsCallback((obj)=>{});
```
回调中的obj为在线参数对象，用户可直接使用，如`obj.pfuSdkTestMode`。**Sdk中一些和在线参数有关的接口需要在此回调中调用。**
### 更多游戏列表
将`pfusdk/prefabs/pfuGamebox.prefab`拖入场景中适当位置。游戏启动后列表会自动初始化。如果玩家需要控制列表的显示隐藏，可以直接引用此控件，通过`node.active = true`控制。

还提供侧边栏的更多游戏列表，将`pfusdk/prefabs/pfuSidePage.prefab`拖入场景适当位置即可

### 更多游戏按钮
游戏中可设置左右两个更多游戏按钮。在UI上创建任意Sprite控件，调整到适当大小。调用如下函数(在**在线参数回调**中调用)
```
var pfuSdk = require("PfuSdk");
pfuSdk.Instance.setMoreGame(this.leftSprite,this.rightSprite);
```
其中leftSprite和rightSprite为两个cc.Sprite的引用,用户可以只传递一个
```
properties: {
        leftSprite:cc.Sprite,
        rightSprite:cc.Sprite,
    },
```
<div id="showShareOrVideo"></div>

### 显示视频
>showVideo

| 参数 | 含义 |
| :------| :------ |
|videoPlacement| 广告位|
|justWatch| 直接看视频(跳过所有参数控制)|
|success| 成功回调|
|fail |失败回调|

>isShareCheckbox 判断当前是否要显示分享勾选框

```
 //进入界面时设置分享勾选框是否可见
 shareTrigger.node.active = PfuSdk.Instance.isShareCheckbox();
 //播放视频时可以把分享框勾选状态传递进去
 PfuSdk.Instance.showVideo({
            justWatch:!shareTrigger.isChecked,
            success:()=>{
                console.log("视频播放成功");
            },
            fail:()=>{
                console.log("视频播放失败");
            }
        })
```
<div id="showShare"></div>

### 分享
>showShare

| 参数 | 含义 |
| :------| :------ |
|shareParams |分享参数
|title |自定义分享标题
|imageUrl |自定义分享图片
|success |成功回调
|fail |失败回调
```
PfuSdk.Instance.showShare({
            success:()=>{
                self.lbShare.string = "分享成功!";
            },
            fail:()=>{
                self.lbShare.string = "分享失败!";
            }
        })
```

### Banner复活
>showBannerRelive

| 参数 | 含义 |
| :------| :------ |
|success |成功回调
|fail |失败回调
```
PfuSdk.Instance.showBannerRelive({
            success:()=>{
                //成功
            },
            fail:()=>{
                //失败
            }
        })
```
### Banner广告
配置`PfuConfig.js`中的`wxBannerId`，Banner广告即可自动创建和显示。配置`bannerHeight`可指定banner高度，对应cocos中的设计像素范围230-170。
主动刷新Banner，调用后不是必然会刷新，具体刷新规则由后台参数控制。
```
pfuSdk.Instance.refreshBanner();
```

如需要自己控制Banner的显示隐藏，请调用HideBanner方法
```
pfuSdk.Instance.HideBanner(true);//隐藏Banner
pfuSdk.Instance.HideBanner(false);//显示Banner
```

### 复活
引导玩家跳转到游戏盒子后复活
```
 PfuSdk.Instance.jumpGameboxForRelive(()=>{
            console.log("复活成功");
        })
```

### 客服功能
将`prefabs/PfuBtnKefu.prefab`拖入任意界面  配置Config中的gameName字段即可

### 获取玩家邀请的新用户列表
使用sdk中的分享功能后，通过分享卡片进入游戏的新用户将被服务器记录。游戏中可以直接使用此邀请用户列表。为了记录玩家的基本信息，游戏必须在第一个界面（未进入主场景前）获取相应权限，相关代码如下：

```
let self = this;
let userInfo = cc.sys.localStorage.getItem("wxUserInfo");
if(!userInfo){
    let button = wx.createUserInfoButton({
    type: 'text',
    text: '获取用户信息',
    style: {
        left: 10,
        top: 76,
        width: 200,
        height: 40,
        lineHeight: 40,
        backgroundColor: '#ff0000',
        color: '#ffffff',
        textAlign: 'center',
        fontSize: 16,
        borderRadius: 4
        }
    })
    button.onTap((res) => {
        console.log(res.userInfo)
        if(res.userInfo){
            cc.sys.localStorage.setItem("wxUserInfo",JSON.stringify(res.userInfo));
        }
    })
}

```
以上代码只会获取一次用户信息，成功后存储在本地，之后不会重新获取。

#### 获取邀请列表
```
let inviteList = pfuSdk.Instance.getInviteUserInfoList();
```
列表中元素对象结构
```
let userInfo = {
    uid:"xxx",//玩家uid
    name:"匿名用户",//昵称
    picUrl:"",//玩家头像地址
    getGift:false,//true:已经领取过邀请奖励，false:未领取过邀请奖励
}
```
其中picUrl为头像地址，用户可使用sdk提供的接口直接将url设置到Sprite上
```
pfuSdk.Instance.setSpriteByUrl(sprite,url);
```
某一邀请用户的邀请奖励被领取后，调用下面接口将其标记为已领取
```
pfuSdk.Instance.onInviteFriendGetGiftFinish(uid);
```

ChangeLog.md
### v0.3.0
* 将红包功能修改为福袋，详情查看[这里](https://github.com/LostChrs/PFUSDK/blob/master/Redpacket.md)
* 增加了客服功能，将`prefabs/PfuBtnKefu.prefab`拖入任意界面  配置Config中的gameName字段即可
### v0.2.7
* 修改了红包的金额规则(去掉了登陆红包)
* 提供了获取banner坐标的方法`getBannerTop`,用来适配UI紧贴Banner的情况
### v0.2.6
* 增加了banner刷新失败的回调 `refreshBanner(failCb)`,用于处理没有Banner的情况
* 增加了获取用户时长的接口 `getPlayTime`
### v0.2.4
* 修改了交叉推广跳转逻辑
* 添加了分享失败弹窗文字的后台控制

### v0.2.3
* 优化SDK时长统计

### v0.2.2
* 分享失败后会弹出确认对话框

### v0.2.1
* banner复活增加了后台开关，关闭的时候会调用视频
* 修改了红包的弹出规则

### v0.1.4
* 增加了点击banner复活的功能showBannerRelive
* 为了避免困惑，将`showShareOrVideo`接口移除，改为`showVideo`。并且增加了新的参数`justWatch`,用来指定是否跳过参数控制直接播放视频
### v0.1.3
* 增加了红包在线参数，审核模式和关闭红包时，红包按钮隐藏
* 修改banner刷新相关规则,增加了主动刷新banner的接口`refreshBanner`
### v0.1.2
* 增加了红包功能，请查看[这里](https://github.com/LostChrs/PFUSDK/blob/master/Redpacket.md)
### v0.1.1
* 修改了更多游戏按钮跳转规则
### v0.1.0
* 修改了`showShare`和`showShareOrVideo`的参数传递方式，请查看[这里](https://github.com/LostChrs/PFUSDK/blob/master/README.md#showShare)
* 移除了单独显示视频的方法`showVideo`,请使用`showShareOrVideo`替换
### v0.0.10
* 增加了获取公众号图片地址的方法`getOfficialAccount`
* 增加了固定banner高度的参数`bannerHeight`，范围为cocos设计分辨率230-170
* 删除了banner大小参数`bannerSize` 
* 修改了更多游戏按钮的跳转规则
### v0.0.9
* 增加了banner刷新时间后台参数
* 修改了视频前弹分享的规则

### v0.0.8
* 优化了更多游戏列表的风格
* 修改了底层的一些统计规则

### v0.0.7
* 增加了自动判断显示分享还是显示视频的接口`isShowShareBtn()`,和`showShareOrVideo(cb,videoPlacement,shareParmas)`

### v0.0.6
* 分享接口增加了自定义标题和图片的参数 `showShare(cb,parmas,title,imageUrl)`
* 增加了分享来源统计
* 增加了侧边栏更多游戏列表

### v0.0.5
* 添加跳转游戏盒子复活功能 `jumpGameboxForRelive()`
### v0.0.4
* 非审核模式下，视频播放失败时会调用分享
* 增加了分享时的数据统计
* 添加了一个在线参数`pfuSdkShareTime`用来控制分享成功的间隔时间
* 增加了判断刘海屏的函数`isIphoneX()`,和普通全面屏的函数`isFullScreen()`

### v0.0.3
* 修复了大量切换场景时导致的bug

pfusdk/PfuConfig.js

let config = {
    version:"1.0.0",//*必填  游戏参数版本号，用来控制运营后台参数
    gameName:"哈哈哈哈",//如需使用客服功能，请配置游戏名称
    bannerHeight:260,//230-170   
    bannerOffYForIpx:30,//在ipx上banner的偏移量，因为不同引擎版本有所区别，所以提供了修改的接口。
    payAppId:"1450018202",//虚拟支付Id，当接入虚拟支付的时候需要填写
    openInviteListListner:false,//开启邀请用户的实时监听
    bannerReliveId:"",
    ipxBanner:70,//微调ipx偏移值

    /*
    * 以下参数为后台自动生成
    */
    appId:1306146,//*必填  嘉丰永道appId
    wxId:"wxc3de0b459142bb1b",//*必填  微信小游戏Id
    privateKey:"bp11lflsecbme9itsb4f",//*必填  嘉丰永道分配给合作方的私钥
    videoId:"adunit-c758250e9f75430c",//视频广告Id
    bannerId:"adunit-8c19e95d3c8e9a7a",//banner广告Id
    navigateToMiniProgramAppIdList:[
        "wxe675b6aad9612c74",
        "wx2d47467291703ec7",
        "wx716b36314be3fe89",
        "wx8b25b991dcc6edf6",
        "wx99e08aff982f8dde",
        "wx3b33f72ee2ec7bc1",
        "wx989199f7dc0e3a50",
        "wxbbbef0244fa4d4e4",
        "wx87fe1890ea1384a5",
        "wxa0eee1fe564aa730"
    ]
};

module.exports = config;

